python/mxnet/gluon/estimator/estimator.py

```diff
@@ -236,32 +231,128 @@ def fit(self, train_data,
                 # update metrics
                 for metric in self.metrics:
                     metric.update(label, pred)
-                    self.train_stats['batch_' + metric.name] = metric.get()[1]
+                    # get metric name and current value and update train stats
+                    self.train_history.set_batch_stats(*metric.get())
                 for loss, loss_metric, in zip(losses, self.loss_metrics):
                     loss_metric.update(0, [l for l in loss])
-                    self.train_stats['batch_' + loss_metric.name] = loss_metric.get()[1]
+                    self.train_history.set_batch_stats(*loss_metric.get())
 
-                try:
-                    self.train_stats['step'] = "{}/{}".format(batch_size * (i + 1), len(train_data._dataset))
-                except AttributeError:
-                    self.train_stats['step'] = i
+                self.train_history.batch_idx = i
+                # record trained samples v.s. total samples if using Gluon DataLoader
+                if isinstance(train_data, gluon.data.DataLoader):
```

Comment:
You might need to rebase this line with the fit-api branch again. 
https://github.com/apache/incubator-mxnet/blob/fit-api/python/mxnet/gluon/estimator/estimator.py#L245